server/src/main/java/io/crate/analyze/AlterTableDropColumnAnalyzer.java

```diff
@@ -0,0 +1,124 @@
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.CoordinatorTxnCtx;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Reference;
+import io.crate.metadata.Schemas;
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.metadata.table.Operation;
+import io.crate.sql.tree.AlterTableDropColumn;
+import io.crate.sql.tree.DropColumnDefinition;
+import io.crate.sql.tree.Expression;
+
+class AlterTableDropColumnAnalyzer {
+
+    private final Schemas schemas;
+    private final NodeContext nodeCtx;
+
+    AlterTableDropColumnAnalyzer(Schemas schemas,
+                                 NodeContext nodeCtx) {
+        this.schemas = schemas;
+        this.nodeCtx = nodeCtx;
+    }
+
+    public AnalyzedAlterTableDropColumn analyze(AlterTableDropColumn<Expression> alterTable,
+                                                ParamTypeHints paramTypeHints,
+                                                CoordinatorTxnCtx txnCtx) {
+        if (!alterTable.table().partitionProperties().isEmpty()) {
+            throw new UnsupportedOperationException("Dropping a column from a single partition is not supported");
+        }
+
+        DocTableInfo tableInfo = (DocTableInfo) schemas.resolveTableInfo(
+            alterTable.table().getName(),
+            Operation.ALTER,
+            txnCtx.sessionSettings().sessionUser(),
+            txnCtx.sessionSettings().searchPath());
+
+        var exprAnalyzerWithFieldsAsString = new ExpressionAnalyzer(
+            txnCtx, nodeCtx, paramTypeHints, FieldProvider.TO_LITERAL_VALIDATE_NAME, null);
+        var exprCtx = new ExpressionAnalysisContext(txnCtx.sessionSettings());
+
+        List<DropColumnDefinition<Expression>> tableElements = alterTable.tableElements();
+
+        List<DropColumnDefinition<Symbol>> addColumnDefinitions = tableElements.stream().map(tableElement -> new DropColumnDefinition<>(
+            exprAnalyzerWithFieldsAsString.convert(tableElement.name(), exprCtx),
+            tableElement.ifExists()
+        )).toList();
+
+        AnalyzedTableElements<Symbol> analyzedTableElements = TableElementsAnalyzer.analyze(
+            List.copyOf(addColumnDefinitions), tableInfo.ident(), tableInfo, false);
+
+        validate(tableInfo, analyzedTableElements);
+        return new AnalyzedAlterTableDropColumn(tableInfo, analyzedTableElements);
+    }
+
+    private static void validate(DocTableInfo tableInfo, AnalyzedTableElements<Symbol> analyzedTableElements) {
+        for (var colIdentToDrop : analyzedTableElements.columnIdents()) {
+            var refToDrop = tableInfo.getReference(colIdentToDrop);
+            if (refToDrop == null) {
+                throw new ColumnUnknownException(colIdentToDrop, tableInfo.ident());
+            }
+
+            if (tableInfo.primaryKey().contains(refToDrop.column())) {
+                throw new UnsupportedOperationException("Dropping column: \"" + colIdentToDrop.sqlFqn() + "\" which " +
```

Comment:
not sure if we want to wrap the col name in `"`. what do you think?

Reply:
I think we can drop it, in case column name already has quotes we would have double quotes which would be confusing?